### PR TITLE
feat: fix `.replace()` when searching string, add `.replaceAll()`

### LINKED ISSUE #221: Unexpected `.replace` behavior
```js
> new (require('magic-string'))('foo+bar').replace('foo\\.bar', 'replaced').toString()
'foo+bar'
> new (require('magic-string'))('foo+bar').replace('foo.bar', 'replaced').toString()
'replaced'
```

This behavior is unexpected to me. Why don't we use `String#indexOf` instead of `String#match`?

### PR DESCRIPTION
**Pull request by fisker**
Fix #221

### CODE REVIEW

File: test/MagicString.js [hunk 1]
```diff
@@ -1305,7 +1305,28 @@ describe('MagicString', () => {
 
 			s.replace('2', '3');
 
-			assert.strictEqual(s.toString(), '1 3 1 2');
+			assert.strictEqual(s.toString(), '1 3 1 3');
```
Comment (fisker) [inline test/MagicString.js:1308]: If we don't want add `.replaceAll`, I guess string replace should replace them all.

**Comment by fisker** [comment]
@antfu Can you take a look?

File: src/MagicString.js [hunk 2]
```diff
@@ -780,4 +780,25 @@ export default class MagicString {
...
+			index = original.indexOf(string, index + stringLength)
+		) {
+			this.overwrite(index, index + stringLength, replacement);
+		}
+
+		return this;
+	}
+
+	replace(searchValue, replacement) {
+		return this[typeof searchValue === 'string' ? '_replaceString' : '_replaceRegexp'](
```
Comment (antfu) [inline src/MagicString.js:799]: Can we use an if condition instead of this? Thanks.

File: test/MagicString.js (hunk 1 above)
Comment (antfu) [reply test/MagicString.js:1308]: I guess it makes better sense to align with JavaScript's string, so I'd prefer adding `.replaceAll` and keeping `.replace` only replaced once.

File: src/MagicString.js (hunk 2 above)
Comment (fisker) [reply src/MagicString.js]: Done

File: test/MagicString.js (hunk 1 above)
Comment (fisker) [reply test/MagicString.js:1308]: Sure!

**Comment by fisker** [comment]
All review point addressed.

**Comment by antfu** [comment]
Awesome, thanks!

**Comment by Tal500** [comment]
Should have added "replaceAll" to the typescript definition file!

**Pull request merged**